This file is part of MXE.
See index.html for further information.

From e91b3a8a4edb0d574f0c71372eddf9889c13dc8e Mon Sep 17 00:00:00 2001
From: Timothy Gu <timothygu99@gmail.com>
Date: Sun, 31 Aug 2014 13:57:14 -0700
Subject: [PATCH 1/2] build: Modernize

dist-lzma was removed in automake 1.11.2.

Signed-off-by: Timothy Gu <timothygu99@gmail.com>

diff --git a/configure.ac b/configure.ac
index 5a2f76c..288031a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -5,13 +5,13 @@ AC_PREREQ([2.63])
 AC_INIT([libbs2b], [3.1.0], [boris_mikhaylov@users.sourceforge.net])
 AC_CONFIG_AUX_DIR([build-aux])
 AM_INIT_AUTOMAKE([1.10.1 -Wall foreign subdir-objects
-                  dist-zip dist-bzip2 dist-lzma])
+                  dist-zip dist-bzip2 dist-xz])
 AC_CONFIG_SRCDIR([src/bs2b.h])
 
 # Checks for programs.
 AC_PROG_CXX
 AC_PROG_CC
-AC_PROG_LIBTOOL
+LT_INIT
 PKG_PROG_PKG_CONFIG
 
 # Checks for libraries.
-- 
1.9.1


From 38e7d5e2641c79c4f34a4731f3f68d60865a9a6b Mon Sep 17 00:00:00 2001
From: MXE
Date: Sun, 31 Aug 2014 14:11:44 -0700
Subject: [PATCH 2/2] Make libsndfile optional

Signed-off-by: Timothy Gu <timothygu99@gmail.com>

diff --git a/configure.ac b/configure.ac
index 288031a..44b3340 100644
--- a/configure.ac
+++ b/configure.ac
@@ -16,8 +16,10 @@ PKG_PROG_PKG_CONFIG
 
 # Checks for libraries.
 PKG_CHECK_EXISTS([sndfile], [], [
-    AC_MSG_ERROR(Please install libsndfile.)
-])
+    AC_MSG_WARN(libsndfile not found. Program building disabled.)
+], [enable_programs=yes])
+
+AM_CONDITIONAL(PROGRAMS, test "x$enable_programs" = "xyes")
 
 # Checks for header files.
 AC_CHECK_HEADERS([fcntl.h malloc.h string.h])
diff --git a/src/Makefile.am b/src/Makefile.am
index 414aeec..cb46f1a 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -1,11 +1,6 @@
 lib_LTLIBRARIES = \
 	libbs2b.la
 
-bin_PROGRAMS = \
-	bs2bconvert \
-	bs2bstream
-
-
 bs2b_HEADERS = \
 	bs2b.h \
 	bs2bclass.h \
@@ -24,6 +19,11 @@ libbs2b_la_SOURCES = \
 	bs2b.c \
 	bs2bclass.cpp
 
+if PROGRAMS
+bin_PROGRAMS = \
+	bs2bconvert \
+	bs2bstream
+
 bs2bconvert_LDADD = \
 	libbs2b.la
 
@@ -38,3 +38,4 @@ bs2bstream_LDADD = \
 
 bs2bstream_SOURCES = \
 	bs2bstream.c
+endif
-- 
1.9.1

